lib/pengine/unpack.c

```diff
@@ -3686,11 +3689,16 @@ remap_operation(struct action_history *history,
                             last_change_s);
                 free(last_change_s);
             }
-            history->rsc->role = RSC_ROLE_PROMOTED;
+            if (!expired
+                || (history->exit_status == history->expected_exit_status)) {
+                history->rsc->role = RSC_ROLE_PROMOTED;
```

Comment:
An expected exit status is a success (i.e. read as "if action was successful or an unexpired failure"). The success check is redundant since successful results don't expire, but that isn't determined in this function, so I didn't want to assume it in case of future changes.

Reply:
Ahhh yeah